src/gui/editorwidgets/qgstexteditwrapper.cpp

```diff
@@ -347,7 +347,9 @@ void QgsTextEditWrapper::setWidgetValue( const QVariant &val )
       mPlainTextEdit->setPlainText( v );
   }
 
-  if ( mLineEdit && val != value() )
+// allow replacement of invalid variants with
+// what was there before - e.g for JSON
+  if ( mLineEdit && ( val != value() || !val.isValid() ) )
```

Comment:
Will this same fix apply to the `mPlainTextEdit` logic above?

Reply:
Yes, well caught, thanks